src/google-classroom-extractor/edfi_google_classroom_extractor/request.py

```diff
@@ -43,14 +43,26 @@ def request_all(
     """
     courses_df: DataFrame = request_all_courses_as_df(classroom_resource, sync_db)
     course_ids: List[str] = courses_df["id"].tolist()
+    students_df = request_all_students_as_df(classroom_resource, course_ids, sync_db)
+    teachers_df = request_all_teachers_as_df(classroom_resource, course_ids, sync_db)
+    usage_df = DataFrame()
+    courseworks_df = DataFrame()
+    submissions_df = DataFrame()
+
+    if arguments.extract_activities:
+        usage_df = request_all_usage_as_df(
+            reports_resource, sync_db, arguments.usage_start_date, arguments.usage_end_date
+        )
+
+    if arguments.extract_assignments or arguments.extract_activities:
```

Comment:
I missed this earlier - why download coursework and submissions when extracting activities?

Reply:
![image](https://user-images.githubusercontent.com/9158083/117368444-c9d29480-ae80-11eb-98d2-e1331aee5fba.png)
If you look at the facade on line 125, it uses submission information for user activities.

Reply:
OK, this is confusing isn't it? We need to add a comment here explaining this situation...  that some assignment submissions are treated as "section activities", and therefore we need to pull all of the submissions when creating "activities" files.